extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -412,6 +419,12 @@ public boolean publishSegments(Set<DataSegment> segments, Object commitMetadata)
             throw new ISE("WTF?! Driver attempted to publish invalid metadata[%s].", commitMetadata);
           }
 
+          // Upgrade TaskLocks for all segments
+          for (DataSegment segment: segments) {
+            toolbox.getTaskActionClient().submit(new SetLockCriticalStateAction(segment.getInterval(),
+                                                                                TaskLockCriticalState.UPGRADE));
```

Comment:
It does acquire locks as it announces, although it's possible that it will announce a segment under a "wider" lock (like announce an hourly segment under a daily lock)